src/peft/tuners/lora.py

```diff
@@ -580,15 +579,22 @@ def _unload_and_optionally_merge(self, merge=True, progressbar: bool = False):
 
         return self.model
 
-    def add_weighted_adapter(self, adapters, weights, adapter_name, combination_type="svd"):
+    def add_weighted_adapter(
+        self, adapters, weights, adapter_name, combination_type="svd", svd_rank=None, svd_params={}, svd_clamp=None
+    ):
         """
         This method adds a new adapter by merging the given adapters with the given weights.
 
         Args:
             adapters (list): List of adapter names to be merged.
             weights (list): List of weights for each adapter.
             adapter_name (str): Name of the new adapter.
-            combination_type (str): Type of merging. Can be one of [`svd`, `linear`]
+            combination_type (str): Type of merging. Can be one of [`svd`, `linear`, `cat`]
+            svd_rank (int): Rank of output adapter for svd. If None provided, will use max rank of merging adapters.
+            svd_params (dict): Additional parameters passed to torch.linalg.svd.
+            svd_clamp (float):
```

Comment:
Same, default can be `None`. I wonder: Should we keep the default of 0.99 for backwards compatibility? If not clamping is clearly the better default, I guess we can change it, since the method was only added recently.

Reply:
I am not sure, for Stable Diffusion LoRAs mixing the provided threshold for 0.99 is not enough (IMO), as the results seem to differ dramatically. So, either the threshold should be bumped up to at least 0.999, or we should provide the ability to disable clamping at all.

So, adding it as a parameter seems a good compromise. Unfortunately, I cannot test right now, how it influences text generation LoRAs, but from my perspective, nothing implicit should be done by default, so `None` is a better option by default.

@pacman100 would you mind if we disable clamping by default?

Reply:
Yes, let's disable the clamping by default